This is a PR with all the features and bugfixes added so far by me and others

tonix-tuft opened the pull request:
Hello guys,

These are all the PRs we have already merged into my two forks (https://github.com/tonix-tuft/tempusdominus and https://github.com/tonix-tuft/tempusdominus-bootstrap)
which I am sending to the main repo of Eonasdan directly on `master`.

I wanna thank @HaroldPutman, @pablo-lp, @zondaTW for their contribution.

Eonasdan commented:
hi @tonix-tuft thanks for doing this. Let's get this bad boy in. This repo will be the base for my revitalization efforts

tonix-tuft commented:
@Eonasdan Great! So, we have changed several things (updated packages, added patches, bugfixes and features), here is a brief summary of what we did:

- Updated composer.json
- Changed composer.json to point to the forked repo; Changed CONTRIBUTING.md for contribution notes - Anton Bagdatyev (Tonix)
- Updated package.json's description
- Added setMultiDate function
- Added Feathers icons support (fallback to Font Awesome icons in case the Feathers icons are not used)
- Patch for locale when using useCurrent option set to false (requires tempusdominus update)
- Bugfix for parseInputDate option
- Bugfix for view update and onUpdate's change property of event payload
- Bugfix for picker when useCurrent option is set to false
- Added setting of isDateUpdateThroughDateOptionFromClientCode flag when modifying the value of date option from the client code
- Fixed styles for picker with calendarWeeks option set to true
- Bugfix for allowInputToggle option when used with input[data-toggle="datetimepicker"]
- Fixed some styles in time picker
- Bugfix for incrementing/decrementing hours/minutes/seconds in multidate mode
- Added CSS minification
- Added watch build command for development
- Downgraded moment version because of issue #5472 https://github.com/moment/moment/issues/5472
- Fixed blinking cursor (UX) on click of a day cell on the calendar in a cross-browser manner
- Changed build scripts
- Updated build commands
- Added multidate mode fixes
- Added bugfixes for .isAfter() and .clone() when picker is in multidate mode
- Added fork section in README
- Fix for issue #325 - respecting ESLint one-var rule
- Added .prettierignore to ignore the whole codebase (for now)
- Updated NPM packages
- Added .prettierignore
- Fix for issue #325

All these things now are on my forked tonix-tuft/tempusdominus-bootstrap repo. How do you want to proceed? Let me know. Thank you!

Eonasdan commented:
@tonix-tuft I'm working on resolving the merge conflicts. There's a couple of other things I'm trying to get resolved right now,  but I  do plan to get this in one way or another. I think I'll put in a PR for the conflicts resolved and then give you a chance to look over it and see if you think anything is missing.

I also want to remove or at least make less embedded the feather icon stuff once it goes in. Dayjs has a plugin model that I think might be worth copying so if FA isn't your thing and uses a different implementation that could be a plugin. I think that might let us do RTL and calendars outside of Georgian.

tonix-tuft commented:
Sounds good! Let me know if you need help with the PR, will be happy to
help whenever I can!

On Fri, Nov 13, 2020, 20:24 Eonasdan <notifications@github.com> wrote:

> @tonix-tuft <https://github.com/tonix-tuft> I'm working on resolving the
> merge conflicts. There's a couple of other things I'm trying to get
> resolved right now, but I do plan to get this in one way or another. I
> think I'll put in a PR for the conflicts resolved and then give you a
> chance to look over it and see if you think anything is missing.
>
> I also want to remove or at least make less embedded the feather icon
> stuff once it goes in. Dayjs has a plugin model that I think might be worth
> copying so if FA isn't your thing and uses a different implementation that
> could be a plugin. I think that might let us do RTL and calendars outside
> of Georgian.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tempusdominus/bootstrap-4/pull/340#issuecomment-726987563>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNKHFNE3OISJF22IXWAO2DSPWBWXANCNFSM4OGM5DWA>
> .
>

Eonasdan commented:
@tonix-tuft pr #360 is in. I had to change node-sass for dart sass as node-sass didn't seem to work on windows 10. I took most of your changes, except I put back some of the branding. I also updated most of the npm packages. I see the momentjs might still be an issue even with the latest. Can you check your specific issue?

Once 360 is merged, I'm going to start in on bringing core and bs 4 into the old repo. Did you make changes to the core project?

Also, does this PR close a bunch of the existing PRs? I don't want to take those changes if it does.

Thanks for your help :)

tonix-tuft commented:
@Eonasdan

> I see the momentjs might still be an issue even with the latest. Can you check your specific issue?

I downgraded moment.js in my fork to stick to version `~2.24.0` (`2.24.*`) because at the time I couldn't compile the project when using moment `>=2.25.0`. You said you tried to upgrade it, which issue did you get?

> Once 360 is merged, I'm going to start in on bringing core and bs 4 into the old repo. Did you make changes to the core project?

Yes I have forked and modified your core repo too (here is the fork: https://github.com/tonix-tuft/tempusdominus).
My PR for core is here: https://github.com/tempusdominus/core/pull/26

> Also, does this PR close a bunch of the existing PRs? I don't want to take those changes if it does.

This PR includes some of the PR you received from the other guys, yes. Here are the ones they resent to my fork and that have already been merged in this PR:

- https://github.com/tonix-tuft/tempusdominus-bootstrap/pull/3 - Feathers Icons Support by @pablo-lp - Originally sent here: https://github.com/tempusdominus/bootstrap-4/pull/335

- https://github.com/tonix-tuft/tempusdominus-bootstrap/pull/4 - Add setMultiDate function by @zondatw - Originally sent here: https://github.com/tempusdominus/bootstrap-4/pull/298

- https://github.com/tonix-tuft/tempusdominus-bootstrap/pull/9 - Return element itself if no target is found by @awh-tokyo - Originally sent here: https://github.com/tempusdominus/bootstrap-4/pull/306

On the core repo, I have received and merged the following PRs:

- https://github.com/tonix-tuft/tempusdominus/pull/1 - Fix moment deprecation warning by @HaroldPutman

- https://github.com/tonix-tuft/tempusdominus/pull/2 - Avoid `Cannot read property 'icons' of undefined` by @Starzu

tonix-tuft commented:
Hi @Eonasdan! I don't use Twitter, I wrote you an email.
